railties/test/generators/controller_generator_test.rb

```diff
@@ -92,6 +92,22 @@ def test_template_engine_with_class_path
     assert_file "app/views/admin/account"
   end
 
+  def test_actions_dont_repeated
+    run_generator ["account", "foo", "bar", "foo"]
```

Comment:
I can see the problem you're working to solve here, as I was able to replicate it in development by running:

`rails generate controller CreditCards foo bar foo`.

You don't see a duplication of views, but you do see duplication in `config/routes` and in the controller actions. Good catch!

Unfortunately, running this test locally against rails master passed on my machine, whereas it should fail. 

What's missing here so that we can have this test cover these changes?

Also, I haven't tested your fix yet locally, but after the test is updated I'm more than happy to! :)

Reply:
Good catch! There a trouble to test this condition because in the `run_generator` helper there a |= assignment for some parameters for the array

https://github.com/rails/rails/blob/c81af6a/railties/lib/rails/generators/testing/behaviour.rb#L67

Maybe should we'll need to move to +| operator instead or by checking if the value too is already inside the `args` arrays for |+ operators.


What do you think about that?

Reply:
I just tried this on my master, and it works as you'd expect it to! :) 

I think there's a good argument for that change as well if you'd like to do it (having me merge into you might be too tricky :) )

Reply:
Thanks a lot to notices this case. I'll moving this change inside this PR.